### ISSUE TITLE
GHM论文理解及实现 | 格物致知

### ISSUE THREAD

**Issue by zheng-yuwei** [zheng-yuwei/my-github-blog-comments#7]
https://zheng-yuwei.github.io/2019/07/08/13_GHM%E8%AE%BA%E6%96%87%E7%90%86%E8%A7%A3%E5%8F%8A%E5%AE%9E%E7%8E%B0/ 

 原论文：Gradient Harmonized Single-stage Detector 本文主要基于tf.keras讨论分类部分，论文也提出了适用于检测的方法。 实验表明具有一定效果，可以尝试，感觉比focal loss要好用。

**Comment by CheneyLu-71** [comment]
GHM这个代码中的valid_mask没有定义是什么意思呀，跑不了代码呢？

**Comment by zheng-yuwei** [comment]
> @CheneyLu-71
> GHM这个代码中的valid_mask没有定义是什么意思呀，跑不了代码呢？
> 

因为我这个是多类别的ghm损失函数，然后在某个样本上有一些类别是无意义的，valid_mask是用来定义哪些类别是有效的。

具体的代码是这个库：https://github.com/zheng-yuwei/multi-label-classification。
有 [issue ](https://github.com/zheng-yuwei/multi-label-classification/issues/5) 说过这个
